controlplane/internal/controllers/scale.go

```diff
@@ -191,6 +191,10 @@ func (r *RKE2ControlPlaneReconciler) removePreTerminateHookAnnotationFromMachine
 	machineOriginal := machine.DeepCopy()
 	delete(machine.Annotations, controlplanev1.PreTerminateHookCleanupAnnotation)
 
+	// Mitigating inssue https://github.com/kubernetes-sigs/cluster-api/issues/11591
```

Comment:
It would be great to create an issue so that we don't forget and once the proper fix lands upstream, we need to remove these lines